evp.go

```diff
@@ -221,12 +222,11 @@ func setupEVP(withKey withKeyFunc, padding int32,
 	h, mgfHash hash.Hash, label []byte, saltLen int32, ch crypto.Hash,
 	init initFunc) (_ _EVP_PKEY_CTX_PTR, err error) {
 	var ctx _EVP_PKEY_CTX_PTR
-	withKey(func(pkey _EVP_PKEY_PTR) int32 {
-		ctx = go_openssl_EVP_PKEY_CTX_new(pkey, nil)
-		return 1
-	})
-	if ctx == nil {
-		return nil, newOpenSSLError("EVP_PKEY_CTX_new failed")
+	if _, err := withKey(func(pkey _EVP_PKEY_PTR) (int32, error) {
+		ctx, err = go_openssl_EVP_PKEY_CTX_new(pkey, nil)
+		return 0, err
```

Comment:
It seems like withKey should just return an error rather than intentionally passing a placeholder (also in `rsa.go`). It looks like `int32, error` happens to line up with what the OpenSSL func returns naturally in a few cases, but not that many.

Reply:
Agree. Will do.